File: HomeHeatManager/src/main/java/com/sasieczno/homeheat/manager/model/HeatingData.java
package com.sasieczno.homeheat.manager.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.LinkedList;

/**
 * The class provides information delivered via management interface
 * by the heat controller process.
 */
@Getter
@Setter
@NoArgsConstructor
public class HeatingData {
    Instant lastMessageTime;
    HeatingPeriod heatingPeriod;
    double externalTemperature;
    double averageExternalTemperature;

    LinkedList<CircuitData> circuits;

    @Getter
    @Setter
    @NoArgsConstructor
    public static class CircuitData {
        int index;
        double temperature;
        boolean heating;
    }
}

File: HomeHeatManager/frontend/src/app/models/heating-data.ts
export class HeatingData {
  controllerStatus: boolean = false;
  lastStatusChangeTime: Date = new Date();
  lastMessageTime: Date = new Date();
  heatingPeriod: string = '';
  externalTemperature: number = 0;
  averageExternalTemperature: number = 0;
  ready: boolean = false;
}
